### ISSUE TITLE
Dataset contains unscaled images which leads to wrong annotation data

### ISSUE THREAD

**Issue by cansik** [CubiCasa/CubiCasa5k#20]
As I understand the dataset, there is always an **original image** of plan as well as a **scaled version**. The **scaled version** can have larger dimensions then the svg, but it is scaled to match the `model.svg` annotation.

Through parsing the plans myself, I found out that some of them are not scaled accordingly. For example `high_quality_architectural/20107` is such a case (and there are a some more).

Both png (original and scaled) are the same size: `w: 485px h: 430px`
While the svg document has a size of `w: 1066px h: 744px`

Even just resizing the png to the svg won't help, because the svg is a bit larger then the matching scaling as you see here:

![image](https://user-images.githubusercontent.com/5220162/87881755-a034a380-c9fb-11ea-88fb-07426b9603ac.png)
*upscale png to svg size*

![image](https://user-images.githubusercontent.com/5220162/87881749-9317b480-c9fb-11ea-839e-dfc7190d21f6.png)
*upscale png to fit svg floorplan*

**Comment by ccmarkus** [comment]
Thanks for this find.
Do you have any additional file paths with this same issue, I could check them out.
Taking in account the data set size, this should not impact in larger scale training that much.

**Comment by cansik** [comment]
It seems there are way less than I thought in the beginning:

```
high_quality_architectural/20107/F1_scaled.png
high_quality_architectural/4884/F1_scaled.png
high_quality_architectural/8002/F1_scaled.png
```

But it seems that there are some rather strange annotations (more is annotated than on the visible plan). While this one seems fine, it gives the door a normalised location value over `1.0`, which leads to errors in further ML pipelines (for example YOLO):

<img src="https://user-images.githubusercontent.com/5220162/88065292-b44be280-cb6c-11ea-8bf8-66b920fb9c0d.png" width="300">

*The black border is from the image tool.*

And here we see even more annotated the on the original plan. This happens on about 2 or three more plans (did not count yet) and can also lead to bugs in further image-classification algorithms:

<img src="https://user-images.githubusercontent.com/5220162/88065416-db0a1900-cb6c-11ea-91f6-ae96e0d2c5f8.png" width="300">

**Comment by anthonymakela** [comment]
I ran into the same issue whilst trying to benchmark with other algorithms. I'm currently in the process of trying to convert the dataset into COCO format for this to be compatible also with different algorithms.

**Comment by cansik** [comment]
I have already written a script for that, but it is quite painful because you have to interpret svg path descriptions and so on.

**Comment by anthonymakela** [comment]
@cansik Yes indeed the svg's tend to be very messy. Would you be kind enough to share the script for converting the dataset?

**Comment by lpasselin** [comment]
@cansik, would you be willing to share your script?
edit: or share the resulting annotation files?

Also, we meet again! (pyk4a)

**Comment by Laudarisd** [comment]
Hello @cansik did you convert annotation to coco format? Is it possible to share the script or converted data with coco format?

**Comment by cansik** [comment]
@lpasselin @Laudarisd I have uploaded my script as a [gist here](https://gist.github.com/cansik/55dc17702f053d1c31076584c6db723c). It expects the dataset in the `./data` folder, but you can change that in main. Also it is possible to add more objects that can be detected to the head of the file, currently only `Door` is active.

This script have been written without the intention to make it public, so please don't expect it to work or be simple to navigate.

**Comment by Nick1288** [comment]
I'm trying to overlay the svg with the png by downscaling the svg to the png's dimensions, but it always deviates, and they are never aligned properly. It isn't just the case with 1 or 2 floorplans, it's all of them. Is there some form of pre-processing i'm missing here to get the svg files aligned properly? Any advice would be extremely appreciated :)
